### DISCUSSION TITLE
Can we use negative Close values in Bollinger Bands?

### DISCUSSION THREAD

**Discussion by M22arius494** [DaveSkender/Stock.Indicators#428] [Q&A] [answered]
Just wanted to report something:

I have an unusual situation where I want to calculate the Bollinger Bands over other values then Crypto prices. And in this case I'm inputting negative values into the close-price. Indicators like MACD and Stochastic are working OK with these values, Its just the BB calculation which does not seem to like negative values.

Is it OK to use negative values? And if you agree that negative values are allowed, can you please check the BB calculation?
I assume its the sum calculation in the GetBollingerBands(), but I have no idea if this is fixable.

Thanks,
Marius

**Comment by DaveSkender** [comment, marked as answer]
Yes.  With an initial review, it seems to be okay with negative numbers, but maybe there are some rare data scenarios that cause a problem.  So far, I don't see any mathematical reason why negative numbers wouldn't work with the formula.

Is it producing Exception type errors or does it just look weird?  Send me some sample problematic quote `history` in CSV format if possible; I'll take a look.

![image](https://user-images.githubusercontent.com/8432125/117547724-1e9c1980-afff-11eb-99ca-2f2976719315.png)

**Reply by M22arius494** [reply]
Thanks for answering Dave, I re-examined the way I add the data to the chart and for some stupid reason I'm checking the SMA. Me (stupid) overlooked that several times. Removing that fixed obviously the weird display.


foreach (var a in resultsbb.Reverse())
            {
                **_if (a.Sma > 0)_**
                {
..
                    seriesbbl.Points.AddXY(a.Date.ToLocalTime(), value);
..
                }
            }

![WrongDisplay](https://user-images.githubusercontent.com/59998287/117549514-14980c00-b03b-11eb-813a-34ac524274ef.png)

**Reply by DaveSkender** [reply]
Ok, I'll say at this point that it is okay to use negative Close values for Bollinger Bands.
